src/features/course-submission/index.tsx

```diff
@@ -0,0 +1,269 @@
+import { DancersRepositoryContext } from 'context/dancer';
+import { ScoresRepositoryContext } from 'context/scores';
+import { AuthenticationRepositoryContext } from 'context/authentication';
+import { DefaultDishGrade } from 'context/dishes/constants';
+import CourseSubmissionDish from './components/course-submission-dish';
+import CourseSubmissionForm from './components/course-submission-form';
+import { CourseSubmissionFormWrapper, CourseSubmissionWrapper } from './styled';
+import { DetailedDishSong, Recipe } from './types';
+import { SongIngredient } from '../submission/types';
+import { DefaultDetailedDishSong, DefaultRecipe } from './constants';
+import { DefaultSongIngredient } from '../submission/constants';
+import { DishSubmissionRequest } from '~/context/dishes/types';
+
+const CourseSubmission = () => {
+  const dishesRepository = useContext(DishesRepositoryContext);
+  const songsRepository = useContext(SongsRepositoryContext);
+  const dancersRepository = useContext(DancersRepositoryContext);
+  const scoresRepository = useContext(ScoresRepositoryContext);
+  const authRepo = useContext(AuthenticationRepositoryContext);
+
+  const loggedInUser = authRepo.authenticationRepositoryInstance
+    .get()
+    .okOrDefault();
+
+  const [form] = Form.useForm();
+
+  const [submitted, setSubmitted] = useState(false);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [currentRecipe, setCurrentRecipe] = useState(DefaultRecipe);
+  const [loading, setLoading] = useState(true);
+  const [sending, setSending] = useState(false);
+  const [recipes, setRecipes] = useState(new Array<Recipe>());
+  const [currentGrade, setCurrentGrade] = useState(DefaultDishGrade);
+
+  const dishRecipeMap = new Map<string, Recipe>();
+  const songIngredientMap = new Map<string, SongIngredient>();
+  const dishSongMap = new Map<string, DetailedDishSong>();
+
+  const onSubmit = async () => {
+    const values = await form.validateFields();
+    setSending(true);
+
+    const request: DishSubmissionRequest = {
+      pairBonus: values.pairBonus,
+      finalImage: values.finalImage,
+      scores: [],
+    };
+    [0, 1, 2].forEach((index) => {
+      request.scores.push({
+        score: values[`score${index}`],
+        scoreImage: values[`scoreImage${index}`],
+        songId: values[`songId${index}`],
+      });
+    });
+
+    const gradeResponse = await dishesRepository.dishesRepositoryInstance.postSubmission(
+      currentRecipe.dish.id,
+      {
+        ...request,
+      },
+    );
+    const grades = await dishesRepository.dishesRepositoryInstance.getGrades(
+      currentRecipe.dish.id,
+    );
+
+    grades.okOrDefault().every((grade) => {
+      if (grade.id === gradeResponse.okOrDefault().gradedDishId) {
+        setCurrentGrade(grade);
+        return false;
+      }
+      return true;
+    });
+
+    form.resetFields();
+    setSubmitted(true);
+    setSending(false);
+  };
+
+  const gradeToInt = (grade: string) => {
```

Comment:
Would it be preferable to return numerical values instead of letter values from the api?